Vertexvis/vertex-web-sdk
# [SDK-1147] Support for double tap and long press events

## What

Adds support for a `doubletap` and `longpress` event. These both default to allowing a second tap to trigger a `doubletap` if it occurs within 500ms, and triggering a `longpress` if a mousedown/touchstart occur and 500ms pass before a mouseup/touchend are received. These durations are configurable through the `viewer.config` property by passing:
```javascript
viewer.config = {
  events: {
    longPressThreshold: {ms},
    doubleTapThreshold: {ms}
  }
}
```

Also made a small adjustment to the touch interaction handler for rotating the camera to allow for single pixel movements after you've initiated a rotation (by moving a distance >= 2 pixels). This allows for more fine-tuned movements of the camera during a touch-based camera interaction.

## Ticket

https://vertexvis.atlassian.net/browse/SDK-1147

## Test Plan

- Test adding a listener for a `doubletap` event
  - When a double tap occurs an event containing the position of the tap should be emitted
- Test adding a listener for a `longpress` event
  - When a long press (mouse/touch held down for 500ms) an event containing the position of the longpress should be emitted
- Test `tap` events
  - Tap events should fire for every tap/click (currently even when a doubletap fires)
- Test configuring the threshold for a `doubletap` event
  - Verify that the double tap can occur during this new threshold
- Test configuring the threshold for a `longpress` event
  - Verify that the long press can occur during this new threshold
- Test touch-based camera manipulation
  - After the initial 2 pixel movement, single pixel movements should manipulate the camera

## Areas of Possible Regression

Tap events, touch-based camera manipulation

## Out of scope changes made in PR

As mentioned above, the logic for when to manipulate the camera on touch events has been slightly adjusted

## Dependencies

N/A

@Vertexvis/sdk 
PTAL
